test/e2e/framework/test_context.go

```diff
@@ -342,7 +342,7 @@ func RegisterClusterFlags(flags *flag.FlagSet) {
 	flags.StringVar(&TestContext.Tooling, "tooling", "", "The tooling in use (kops, gke, etc.)")
 	flags.StringVar(&TestContext.OutputDir, "e2e-output-dir", "/tmp", "Output directory for interesting/useful test data, like performance data, benchmarks, and other metrics.")
 	flags.StringVar(&TestContext.Prefix, "prefix", "e2e", "A prefix to be added to cloud resources created during testing.")
-	flags.StringVar(&TestContext.MasterOSDistro, "master-os-distro", "debian", "The OS distribution of cluster master (debian, ubuntu, gci, coreos, or custom).")
+	flags.StringVar(&TestContext.ControlPlaneOSDistro, "control-plane-os-distro", "debian", "The OS distribution of cluster control plane (debian, ubuntu, gci, coreos, or custom).")
```

Comment:
we should probably support two flag names for some time so we can migrate smoothly, besides the project's own CI there are external CI setting these flags

Reply:
this may also allow us to simplify the migration in kubetest etc., we can probably detect the version (? not sure in all situations if this is true off the top of my head ...) and have kubetest(2) use the old flag for versions older than the first release that contains this change.

Reply:
some more of the places within the project: https://cs.k8s.io/?q=master-os-distro&i=nope&files=&excludeFiles=&repos=

Reply:
so to support two flag names, should both the old and new line be kept?

Reply:
We will need to read the flags into two variables and then error if both are set. If only one is set we use that value for the actual variable.

Reply:
Having two flags until the removal of the master flag seems better. We should also mark it as deprecated - e.g. https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/cmd/kube-controller-manager/app/options/deprecatedcontroller.go#L38

I recommended renaming the flag right away since I recalled a prior case where we claimed the suite CLI to not have support guarantee for > 1 releases. I don't remember what it was about but `@liggitt ` might have participated in the discussion.

Reply:
So just to make sure what needs to be done: master-os-distro needs to be added back to test_context, anywhere that this flag is referenced will need to check for both flags and check which one is valid, and master-os-distro needs to be marked as deprecated in deprecatedcontroller. Please let me know if this is incorrect.

Reply:
> I recommended renaming the flag right away since I recalled a prior case where we claimed the suite CLI to not have support guarantee for > 1 releases. I don't remember what it was about but @liggitt might have participated in the discussion.

I think that's fair from a downstream perspective potentially, but for ourselves I think it will be easier to figure out how to make tools like kubetest handle the rename if we can use the kubernetes version to detect support. Otherwise maybe we could process the output of `--help` 🤔 .

> So just to make sure what needs to be done: master-os-distro needs to be added back to test_context, anywhere that this flag is referenced will need to check for both flags and check which one is valid, and master-os-distro needs to be marked as deprecated in deprecatedcontroller. Please let me know if this is incorrect.

No there should not be any change in deprecatedcontroller.go, that link is just an example of marking a flag as deprecated. The flag should be marked deprecated in the e2e sources.

Any approach that solves the problem of "we still need to set this flag in CI, with the project's own CI tooling that must support additional older releases of kubernetes that will not have this change" can be accepted.

I think one of the easiest may be adding a new flag with the new name, marking the old flag deprecated, erroring if both are set, and taking the value in the variable from whichever *is* set. Then the other tools in the project can be updated separately to set the correct flag depending on the version.

Another approach would be proactively updating the other tools to detect the flag in some other way, and then setting the one that is present.

Either way, tools like kubetest must be fixed to handle both the new and old flag depending on which is available, so that CI passses on this change, and so we do not break CI on older Kubernetes versions (e.g. by _only_ setting the new flag in kubetest, which would break older releases).